idk/kafka_sasl/source.go

```diff
@@ -154,32 +171,41 @@ func (r *Record) Schema() interface{} { return nil }
 func (r *Record) Commit(ctx context.Context) error {
 	r.src.mu.Lock()
 	defer r.src.mu.Unlock()
-
 	idx, base := r.idx, r.src.spoolBase
 	if idx < base {
 		return errors.New("cannot commit a record that has already been committed")
 	}
 	section, remaining := r.src.spool[:idx-base], r.src.spool[idx-base:]
-	// sort by increasing partition, decreasing offset
 	sort.Slice(section, func(i, j int) bool {
-		if section[i].Partition != section[j].Partition {
+		if *section[i].Topic != *section[j].Topic {
+			return *section[i].Topic < *section[j].Topic
+		} else if section[i].Partition != section[j].Partition {
```

Comment:
I didn't look at this but I can. I copied this from the `kafka` consumer. At one point Q2 was incorrectly committing offsets with the `kafka` consumer and this was the logic that fixed it.

In general, I tried to move kafka source logic to the kafka_sasl source where it made sense to do so. Since I joined, I haven't known any customers to use the `kafka_sasl` consumer but Q2 has use the `kafka` consumer heavily. So I'm working of the assumptions that the `kafka` consumer is better tested so changes made there sound be made in the `sasl` consumer as well.